java_sdk_example/src/main/java/aws/sample/paymentcryptography/p2pe/PaymentProcessorService.java

```diff
@@ -32,29 +31,35 @@ public class PaymentProcessorService {
     @ResponseBody
     public String authorizePayment(@RequestParam String encryptedData, @RequestParam String ksn) {
         try {
-            AWSPaymentCryptographyData dataPlaneClient = DataPlaneUtils.getDataPlaneClient();
-
-            DukptEncryptionAttributes dukptEncryptionAttributes = new DukptEncryptionAttributes()
-                    .withKeySerialNumber(ksn)
-                    .withMode(ServiceConstants.MODE);
-
-            EncryptionDecryptionAttributes decryptionAttributes = new EncryptionDecryptionAttributes();
-            decryptionAttributes.setDukpt(dukptEncryptionAttributes);
-
-            DecryptDataRequest decryptDataRequest = new DecryptDataRequest();
-            decryptDataRequest.setCipherText(encryptedData);
-            decryptDataRequest.setKeyIdentifier(ServiceConstants.BDK_ALIAS);
-            decryptDataRequest.setDecryptionAttributes(decryptionAttributes);
+            PaymentCryptographyDataClient dataPlaneClient = DataPlaneUtils.getDataPlaneClient();
+
+            DukptEncryptionAttributes dukptEncryptionAttributes = DukptEncryptionAttributes
+                    .builder()
+                    .keySerialNumber(ksn)
+                    .mode(ServiceConstants.MODE)
+                    .build();
+
+            EncryptionDecryptionAttributes decryptionAttributes = EncryptionDecryptionAttributes
+                    .builder()
+                    .dukpt(dukptEncryptionAttributes)
+                    .build();
+
+            DecryptDataRequest decryptDataRequest = DecryptDataRequest
+                    .builder()
+                    .cipherText(encryptedData)
+                    .keyIdentifier(ServiceConstants.BDK_ALIAS)
+                    .decryptionAttributes(decryptionAttributes)
+                    .build();
 
             Logger.getGlobal()
                     .info("PaymentProcessorService:authorizePayment Attempting to decrypt data " + encryptedData
                             + " by AWS Cryptography Service");
-            DecryptDataResult decryptDataResult = dataPlaneClient.decryptData(decryptDataRequest);
+            DecryptDataResponse decryptDataResponse = dataPlaneClient.decryptData(decryptDataRequest);
 
             PKCS7Padding padder = new PKCS7Padding();
-            int padCount = padder.padCount(Hex.decodeHex(decryptDataResult.getPlainText()));
```

Comment:
Recommend putting this into a helper method.

Reply:
done